### PR TITLE
#0

### PR DESCRIPTION
**Pull request by hruiyang**
Rui Yang here!

Any comments on what's wrong with the smart contract would be great!!

Currently there's no access control for burn and mint! (Probably import Ownable from OpenZeppelin and just use it?)

### CODE REVIEW

**Comment by Ddorsmt** [comment]
I see no issues with the contract.
Maybe add a test trying to mint/burn as a non-owner?

**Comment by huy525** [comment]
Everything looks fine now! Maybe test that a non-owner can't mint?

File: contracts/ERC20.sol [hunk 1]
```diff
@@ -1,26 +1,56 @@
...
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract ERC20 is IERC20Metadata, Ownable {
+  uint256 private _totalSupply;
+  uint256 private _maxSupply;
+  string private _name;
+  string private _symbol;
+  uint8 private _decimals;
+
+  mapping(address => uint256) balances;
```
Comment (UncleGrandpa925) [inline contracts/ERC20.sol]: You should always have the visibility modifiers

File: contracts/ERC20.sol [hunk 2]
```diff
@@ -29,6 +59,51 @@ contract ERC20 is IERC20 {
...
     uint256 amount
   ) external returns (bool) {
+    require(from != address(0), "Address [from] is zero");
+    require(to != address(0), "Address [to] is zero");
+    require(balances[from] >= amount, "Insufficient balance to transferfrom");
+    require(allowances[from][msg.sender] >= amount, "Insufficient allowance to transferfrom");
+    if (allowances[from][msg.sender] != type(uint256).max) {
+      allowances[from][msg.sender] -= amount;
+    }
+    _transfer(from, to, amount);
```
Comment (UncleGrandpa925) [inline contracts/ERC20.sol:69]: good call to _transfer function

**Pull request closed without merging**